init_node_script_test.go

```
@@ -80,6 +95,43 @@ printf 'prometheus = false\nprometheus_listen_addr = ":26660"\n' > "$FAKE_HOME/c
+	// wrapper instead of silently producing a partially configured sink.
+	fake := `#!/bin/sh
+set -eu
+mkdir -p "$FAKE_HOME/config"
+printf 'minimum-gas-prices = ""\n[telemetry]\nservice-name = ""\nenabled = false\nenable-hostname = false\nenable-hostname-label = false\nprometheus-retention-time = 0\n[api]\nenable = false\n' > "$FAKE_HOME/config/app.toml"
+printf 'prometheus = false\nprometheus_listen_addr = ":26660"\n' > "$FAKE_HOME/config/config.toml"
+`
+	if err := os.WriteFile(fakeBinary, []byte(fake), 0o700); err != nil {
+		t.Fatal(err)
+	}
+
+	cmd := exec.Command("bash", "scripts/init-node.sh")
```

golangci-lint (2.12.2)

[error] os/exec.Command must not be called. use os/exec.CommandContext

(noctx)